test/boost/locator_topology_test.cc

```diff
@@ -47,21 +48,20 @@ SEASTAR_THREAD_TEST_CASE(test_add_node) {
         set_abort_on_internal_error(true);
     });
 
-    std::unordered_set<const locator::node*> nodes;
-
-    nodes.insert(topo.add_node(id2, ep2, endpoint_dc_rack::default_location, node::state::normal));
-    nodes.insert(topo.add_node(id1, ep1, endpoint_dc_rack::default_location, node::state::normal));
+    std::unordered_set<std::reference_wrapper<const locator::node>> nodes;
+    nodes.insert(std::cref(*(topo.add_node(id2, ep2, endpoint_dc_rack::default_location, node::state::normal))));
```

Comment:
Looks like `topo.add_node` could also return `const node&`, it's weird that we have to `*` it first.

Unless `topo.add_node` can return `nullptr`? But it doesn't seem so -- the implementation of `add_node` assumes that the given pointer is not `nullptr`, and then basically returns a reference to the pointee.